apps/obsidian/src/components/BulkImportModal.tsx

```diff
@@ -0,0 +1,435 @@
+      enabled: false,
+    }));
+    setPatterns(initialPatterns);
+  }, [plugin.settings.nodeTypes]);
+
+  const handlePatternChange = (
+    index: number,
+    field: keyof BulkImportPattern,
+    value: string | boolean,
+  ) => {
+    setPatterns((prev) =>
+      prev.map((pattern, i) =>
+        i === index ? { ...pattern, [field]: value } : pattern,
+      ),
+    );
+  };
+
+  const handleScanVault = useCallback(async () => {
+    const enabledPatterns = patterns.filter(
+      (p) => p.enabled && p.alternativePattern.trim(),
+    );
+
+    if (!queryEngineRef.current) {
+      new Notice("Query engine not initialized");
+      return;
+    }
+
+    setIsScanning(true);
+    try {
+      const validNodeTypes = plugin.settings.nodeTypes;
+      const foundCandidates =
+        await queryEngineRef.current.scanForBulkImportCandidates(
+          enabledPatterns,
+          validNodeTypes,
+        );
+
+      setCandidates(foundCandidates);
+      setStep("review");
+    } catch (error) {
+      console.error("Error scanning vault:", error);
+      new Notice("Error scanning vault for candidates");
+    } finally {
+      setIsScanning(false);
+    }
+  }, [patterns, plugin]);
+
+  const handleCandidateToggle = (index: number) => {
+    setCandidates((prev) =>
+      prev.map((candidate, i) =>
+        i === index
+          ? { ...candidate, selected: !candidate.selected }
+          : candidate,
+      ),
+    );
+  };
+
+  const handleFolderToggle = (folderPath: string) => {
+    setCandidates((prev) => {
+      const folderCandidates = prev.filter(
+        (c) => getDirectoryPath(c.file.path) === folderPath,
+      );
+      const allSelected = folderCandidates.every((c) => c.selected);
+      return prev.map((c) =>
+        getDirectoryPath(c.file.path) === folderPath
+          ? { ...c, selected: !allSelected }
+          : c,
+      );
+    });
+  };
+
+  const handleBulkImport = async () => {
+    const selectedCandidates = candidates.filter((c) => c.selected);
+    setStep("importing");
+    setImportProgress({ current: 0, total: selectedCandidates.length });
+
+    let successCount = 0;
+
+    for (const candidate of selectedCandidates) {
+      try {
+        await plugin.app.fileManager.processFrontMatter(
+          candidate.file,
+          (fm) => {
+            fm.nodeTypeId = candidate.matchedNodeType.id;
+          },
+        );
+
+        successCount++;
+      } catch (fileError) {
+        console.error(
+          `Error processing file ${candidate.file.path}:`,
+          fileError,
+        );
+
+        try {
+          const fileContent = await plugin.app.vault.read(candidate.file);
+          const newContent = `---\nnodeTypeId: ${candidate.matchedNodeType.id}\n---\n\n${fileContent}`;
+
+          await plugin.app.vault.modify(candidate.file, newContent);
+
+          successCount++;
+
+          new Notice(
+            `Error processing ${candidate.file.basename}. Preserved original content.`,
```

Comment:
Let's update this to let the user know we still added new YAML metadata.  Maybe remove the word "Error" and replace it with something less alarming, like "Notice" or "Problem"